lib/hal/video.c

```diff
@@ -387,6 +400,24 @@ void XVideoSetSoftenFilter(BOOL enable)
 	}
 }
 
+void XVideoSetGammaRamp(unsigned int offset, const GAMMA_RAMP_ENTRY *entries, unsigned int count) {
+	int i;
+
+	assert((offset + count) <= 256);
+	memcpy(&gamma_ramp_entries[offset], entries, count * sizeof(GAMMA_RAMP_ENTRY));
```

Comment:
Since you've cached the gamma ramp entries, should we enforce people access only through a corresponding `XVideoSetGammaRamp` function?

Reply:
> should we make `gamma_ramp_entries` static

No, because this also applies to other code in hal/video.c.
We'll have to refactor this. I'll create an issue later if none exists yet.

*(It used to be static before I made style consistent in preparation for this PR)*

>  if necessary later, a corresponding `XVideoGetGammaRamp` function?

This branch used to have this function, but it was a last-minute removal before sending the PR.

I decided to remove it because there's no way to read the initial system state (there might be a way, but I don't know how yet).
So this can leave us in a situation where we are unable to use this function because the gamma ramp was never set.

Particularily, this will become a problem when introducing another video API to replace hal/video, which doesn't automatically set the state (to be able to take over rendering from another XBE; e.g.: menu.xbe &rarr; ingame.xbe where ingame.xbe should do a lightweight video initialization without resetting gamma ramps etc.)

So I'm being a bit cautious with adding an API yet.

Reply:
Fair enough